ci_framework/roles/reproducer/templates/play.yml.j2

```diff
@@ -1,11 +1,93 @@
 ---
+{% if use_content_provider | default(false) | bool -%}
+- name: Configure CRC for insecure registry
+  hosts: crcs
+  vars:
+    content_provider_registry_ip: {{ cifmw_reproducer_ctl_ip4 }}
+  tasks:
+    - name: Set insecure registry on crc node
+      ansible.builtin.import_tasks: ci/playbooks/tasks/set_crc_insecure_registry.yml
+
+- name: "Content Provider"
+  hosts: localhost
+  gather_facts: true
+  vars:
+    cifmw_rp_registry_ip: {{ cifmw_reproducer_ctl_ip4 }}
+    cifmw_rp_registry_firewall: false
+    job_id: "{{ job_id }}"
+  tasks:
+{% raw %}
+    - name: Load env variables
+      ansible.builtin.include_vars:
+        file: "{{ item }}"
+      loop:
+        - "{{ ansible_user_dir }}/{{ job_id }}-params/zuul-params.yml"
+        - "./scenarios/centos-9/base.yml"
+        - "./scenarios/centos-9/content_provider.yml"
+        - "./scenarios/centos-9/zuul_inventory.yml"
+
+    - name: Install necessary tools
+      ansible.builtin.import_role:
+        name: install_yamls_makes
+        tasks_from: make_download_tools
+
+    - name: Deploy registry
+      ansible.builtin.import_role:
+        name: registry_deploy
+
+    # Imported from the ci/playbooks/content_provider/content_provider.yml play
+    # Ensure we get consistent name for the operator, linked to
+    # https://github.com/openstack-k8s-operators/ci-framework/pull/621
+    - name: Set var for cifmw_operator_build_operators var
+      when:
+        - zuul is defined
+        - "'project' in zuul"
+        - "'short_name' in zuul.project"
+      ansible.builtin.set_fact:
+        cifmw_operator_build_operators:
+          - name: "openstack-operator"
+            src: "~/src/github.com/{{ cifmw_operator_build_org }}/openstack-operator"
+            image_base: >-
+              {{ zuul.project.short_name | split('-') | reject('search','operator') | join('-') }}
+
+    - name: Build Operators
+      ansible.builtin.include_role:
+        name: operator_build
+
+    - name: Get the containers list from container registry
+      ansible.builtin.uri:
+        url: "http://{{ cifmw_rp_registry_ip }}:5001/v2/_catalog"
```

Comment:
wondering why we hard-code to 5001 ... it seems to be the case in many places, but grep tells me there is a var for that
https://github.com/openstack-k8s-operators/ci-framework/blob/d071087d8b9c12b00ac3dfaf68dace34d8df30ba/ci_framework/roles/registry_deploy/defaults/main.yml#L24

Reply:
that part was also copied from the pre-existing playbook.

Unfortunately, I can't just run that playbook since it's doing far more than deploying the registry + build content, it also manages logs and related things that I do not want to get at that stage of the run in the reproducer. Maybe refactoring that ci playbook in order to allow a better inclusion is something to consider, in a follow-up.